Applications/Spire/Source/Ui/ListItem.cpp

```diff
@@ -8,23 +8,50 @@ using namespace boost::signals2;
 using namespace Spire;
 using namespace Spire::Styles;
 
+namespace {
+  auto DEFAULT_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0xFF, 0xFF, 0xFF))).
+      set(border(scale_width(1), QColor::fromRgb(0, 0, 0, 0))).
+      set(horizontal_padding(scale_width(8)));
+    style.get(Hover()).set(
+      BackgroundColor(QColor::fromRgb(0xF2, 0xF2, 0xFF)));
+    style.get(Focus()).set(
+      border_color(QColor::fromRgb(0x4B, 0x23, 0xA0)));
+    style.get(Selected()).set(
+      BackgroundColor(QColor::fromRgb(0xE2, 0xE0, 0xFF)));
+    return style;
+  }
+}
+
 ListItem::ListItem(QWidget* component, QWidget* parent)
-    : QWidget(parent) {
+    : QWidget(parent),
+      m_is_selected(false) {
   setFocusPolicy(Qt::StrongFocus);
   auto layout = new QHBoxLayout(this);
   layout->setContentsMargins({});
-  m_button = new Button(component, this);
+  auto box = new Box(component);
+  box->setDisabled(true);
+  m_button = new Button(box, this);
   setFocusProxy(m_button);
   m_button->installEventFilter(this);
   layout->addWidget(m_button);
-  auto style = get_style(*m_button);
-  style.get(Body()).set(
-    BackgroundColor(QColor::fromRgb(0xFF, 0xFF, 0xFF)));
-  style.get(Hover() / Body()).set(
-    BackgroundColor(QColor::fromRgb(0xF2, 0xF2, 0xFF)));
-  style.get(Focus() / Body()).set(
-    BackgroundColor(QColor::fromRgb(0x68, 0x4B, 0xC7)));
-  set_style(*m_button, std::move(style));
+  proxy_style(*this, *box);
+  set_style(*this, DEFAULT_STYLE());
```

Comment:
This is styling the entire component, but the spec is styling only the ```Box```.

Reply:
The style proxy of the ListItem is the Box, so the styling is actually applying on the Box.

Reply:
A proxy is one-way, so styling the ```ListItem``` styles the ```Box``` but styling the ```Box``` does not style the ```ListItem```. This is an important detail since a parent of ```ListItem``` can override ```ListItem```'s style using ```set_stylesheet``` which will completely wipe out all of the intended styling for the ```Box```.